pants-plugins/api_spec/rules.py

```diff
@@ -63,79 +50,39 @@ class APISpecFieldSet(FieldSet):
 
 class GenerateAPISpecViaFmtTargetsRequest(FmtTargetsRequest):
     field_set_type = APISpecFieldSet
-    name = GENERATE_CMD
+    tool_subsystem = GenerateApiSpec
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
 
 
 class ValidateAPISpecRequest(LintTargetsRequest):
     field_set_type = APISpecFieldSet
-    name = VALIDATE_CMD
+    tool_subsystem = ValidateApiSpec
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
 
 
 @rule(
     desc="Update openapi.yaml with st2-generate-api-spec",
     level=LogLevel.DEBUG,
 )
 async def generate_api_spec_via_fmt(
-    request: GenerateAPISpecViaFmtTargetsRequest,
+    request: GenerateAPISpecViaFmtTargetsRequest.Batch,
+    subsystem: GenerateApiSpec,
 ) -> FmtResult:
     # There will only be one target+field_set, but we iterate
     # to satisfy how fmt expects that there could be more than one.
```

Comment:
Does this comment make sense now we have removed this chunk of code?

Reply:
Hmm. Yup. This is out-of-date. I'll drop the comment.